Source/Core/Core/WiiRoot.cpp

```diff
@@ -26,10 +26,11 @@
 
 namespace Core
 {
-static std::string s_temp_wii_root;
-
 namespace FS = IOS::HLE::FS;
 
+static std::string s_temp_wii_root;
+static std::unique_ptr<FS::FileSystem> s_netplay_sync_fs;
```

Comment:
Is there really no way to avoid grafting netplay into this code as a file-static variable, or extending the interface to allow this to live within the NetPlayClient instead?

Reply:
Yeah, given the nature of what I'm doing, there has to be a static variable *somewhere* to hold this object.

Reply:
Note that this adds to the global state problem (which we've already gone through the song and dance of before, mind), and also adds more problems to solve to eventually making Core a proper library long-term (`s_temp_wii_root` is a pre-existing smell that used to exist within Common before being moved to Core to avoid circular dependencies).

Netplay should also not directly graft into core like this with object instances solely intended for it. All necessary object instances, ideally, should be within the relevant netplay source files (also, ideally not static, since it makes it impossible to start more than one instance of the core without trampling data, which is what a proper library should allow). Anything else needed should be a result of changing the existing core APIs to accommodate the difference in behavior.

Reply:
Core as a whole is unfortunately not set up to handle this in a non-static way yet, so until a major refactor, I can't see any reasonable way to avoid doing this for now.

Reply:
Alright, but surely it must be possible to relocate it to netplay source files (without making it an external global), correct?

Reply:
Yeah, I can probably move the variable into the NetPlay files and namespace, if that would be sufficient until Core is actually refactored to be instanced.